File: unit01-hw/areatriangle.rb
#! /usr/bin/ruby
################################################################
# CircleArea.rb
# Computes the area of a triangle.
#
# Input: The height of a triangle.
# Input: The width of a triangle.
# Precondition: The height is a positive number.
# Precondition: The width is a positive number.
# Output: The area of the triangle.
#
# Completed by: Robert Lowman
# Date: 9/7/16
################################################################

HALF = 0.5

# Function that computes the area of a triangle.
#
# Parameter: height The height of the triangle.
# Parameter: width The width of the triangle.
# Precondition: h > 0.
# Precondition: w > 0.
# Returns: The area of the triangle.
def computeArea(h, w) # define a function with 2 parameters
  HALF * h * w # calculate area of triangle
end # end function

# Where execution begins
if __FILE__ == $0 # If file can be read, continue
  print "Enter the height: " # print string
  height = gets.chomp.to_f # get data from keyboard and set to height
  print "Enter the width: " # print string
  width = gets.chomp.to_f # get data from keyboard and set to width
  print "Area is: "  # print string
  puts computeArea(height, width) # print return value from function
end # end execution

File: unit01-hw/areatriangle.cpp
/***************************************************************
 * Computes the area of a triangle.
 *
 * Input: The radius of a triangle.
 * Precondition: The height and width is a positive number.
 * Output: The area of the triangle.
 *
 * Completed by: Robert Lowman
 * Date: 9/7/16
 ***************************************************************/

#include <iostream>     // enables input and output
using namespace std;

/**
 * Function that computes the area of a triangle.
 *
 * Parameter: height The height of the triangle.
 * Parameter: width The width of the triangle.
 * Precondition: height > 0
 * Precondition: width > 0
 * Returns: The area of the triangle.
 */
double computeArea(double height, double width)
{
  return .5 * height * width;            // return an expression
}

/**
 * Main method.
 */
int main()
{
                                          // prompt for and input radius
  cout << "To compute the area of a triangle, enter its height: ";
  double height;                          // variable declaration
  cin >> height;                          // read height from keyboard
  cout << "To compute the area of a triangle, enter its width: ";
  double width;                          // variable declaration
  cin >> width;                          // read width from keyboard
  double area = computeArea(height, width);      // compute area
  cout << "The area is " << area << endl; // output area
}

